### PR TITLE
Add C++11 compatibility 

### PR DESCRIPTION
**Pull request by cloudhan**
Partially Implemented #32 (I don't think support C++03 is possible without a rewrite).

Built with GCC-5/8, C++11/14
Built with VS2019 C++11/14

### CODE REVIEW

**Comment by cloudhan** [comment]
Please note, with VS2019 C++11, there is something weird with the `rh` executable, only a few test cases are listed here:

```powershell
> .\rh.exe -ltc
[doctest] listing all test case names
===============================================================================
move to moved<robin_hood::unordered_node_map<int, int>>
move to moved<robin_hood::unordered_flat_map<int, int>>
swapempty<robin_hood::unordered_node_map<int, int>>
swapempty<robin_hood::unordered_flat_map<int, int>>
clz
copyassignment
include only
pair trivially copy/destructible
rng_is_deterministic
===============================================================================
[doctest] unskipped test cases passing the current filters: 9
```

File: src/include/robin_hood.h [hunk 1]
```diff
@@ -122,6 +122,77 @@
```
Comment (martinus) [inline src/include/robin_hood.h]: Please update the versions at line 38 for this update to 3.3.0:
```cpp
#define ROBIN_HOOD_VERSION_MAJOR 3  // for incompatible API changes
#define ROBIN_HOOD_VERSION_MINOR 3  // for adding functionality in a backwards-compatible manner
#define ROBIN_HOOD_VERSION_PATCH 0 // for backwards-compatible bug fixes
```

File: src/include/robin_hood.h [hunk 2]
```diff
@@ -367,8 +438,16 @@ class BulkPoolAllocator {
...
     // enforce byte alignment of the T's
-    static constexpr size_t ALIGNMENT =
+#if __cplusplus >= 201402L
+    static const size_t ALIGNMENT =
         (std::max)(std::alignment_of<T>::value, std::alignment_of<T*>::value);
+#else
+    static const size_t ALIGNMENT =
+        (ROBIN_HOOD_STD_COMP::alignment_of<T>::value > ROBIN_HOOD_STD_COMP::alignment_of<T*>::value)
+            ? ROBIN_HOOD_STD_COMP::alignment_of<T>::value
+            : +ROBIN_HOOD_STD_COMP::alignment_of<T*>::value; // the + is for walkarround
```
Comment (martinus) [inline src/include/robin_hood.h]: I don't understand the `+` and what do you mean by walkaround?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,16 @@
 cmake_minimum_required(VERSION 3.5)
 project(robin-hood-hashing LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 14)
```
Comment (martinus) [inline CMakeLists.txt]: Can you update the `.travis.yml` to include  c++11 builds as well?

File: src/include/robin_hood.h (hunk 2 above)
Comment (cloudhan) [reply src/include/robin_hood.h]: I will cause a duplicate branch warning, blahblah, and then with "treat warning as error" enabled, caused a problem with my MSVC build.

File: CMakeLists.txt (hunk 3 above)
Comment (cloudhan) [reply CMakeLists.txt]: Done

**Pull request merged**